packages/components/tests/integration/components/hds/form/field/index-test.js

```diff
@@ -0,0 +1,118 @@
+        </Hds::Form::Field>`
+    );
+    assert.dom('#test-form-field .hds-form-field__label').exists();
+    assert.dom('.hds-form-field__label').hasText('This is the label');
+    assert.dom('#test-form-field .hds-form-field__helper-text').exists();
+    assert
+      .dom('.hds-form-field__helper-text')
+      .hasText('This is the helper text');
+    assert.dom('#test-form-field .hds-form-field__control').exists();
+    assert.dom('.hds-form-field__control').hasText('This is a mock control');
+    assert.dom('#test-form-field .hds-form-field__error').exists();
+    assert.dom('.hds-form-field__error').hasText('This is the error');
+  });
+  test('it automatically provides all the ID relations between the elements', async function (assert) {
+    assert.expect(4);
+    await render(
+      hbs`<Hds::Form::Field @layout="vertical" id="test-form-field" as |F|>
+          <F.Label>This is the label</F.Label>
+          <F.HelperText>This is the helper text</F.HelperText>
+          <F.Control><pre class="hds-form-field__control" id={{F.id}} aria-describedby={{F.ariaDescribedBy}}>This is a mock control</pre></F.Control>
+          <F.Error>This is the error</F.Error>
+        </Hds::Form::Field>`
+    );
+    // the control ID is dynamically generated
+    let control = this.element.querySelector(
+      '#test-form-field .hds-form-field__control'
+    );
+    let controlId = control.id;
+    assert.dom('.hds-form-field__label').hasAttribute('for', controlId);
+    assert
+      .dom('.hds-form-field__helper-text')
+      .hasAttribute('id', `helper-text-${controlId}`);
+    assert
+      .dom('.hds-form-field__control')
+      .hasAttribute(
+        'aria-describedby',
+        `helper-text-${controlId} error-${controlId}`
+      );
+    assert
+      .dom('.hds-form-field__error')
+      .hasAttribute('id', `error-${controlId}`);
+  });
+  test('it automatically provides all the ID relations between the elements with a custom @id', async function (assert) {
+    assert.expect(4);
+    await render(
+      hbs`<Hds::Form::Field @layout="vertical" id="test-form-field" @id="my-custom-id" as |F|>
+          <F.Label>This is the label</F.Label>
+          <F.HelperText>This is the helper text</F.HelperText>
+          <F.Control><pre class="hds-form-field__control" id={{F.id}} aria-describedby={{F.ariaDescribedBy}}>This is a mock control</pre></F.Control>
+          <F.Error>This is the error</F.Error>
+        </Hds::Form::Field>`
+    );
+    let controlId = 'my-custom-id';
+    assert.dom('.hds-form-field__label').hasAttribute('for', controlId);
+    assert
+      .dom('.hds-form-field__helper-text')
+      .hasAttribute('id', `helper-text-${controlId}`);
+    assert
+      .dom('.hds-form-field__control')
+      .hasAttribute(
+        'aria-describedby',
+        `helper-text-${controlId} error-${controlId}`
+      );
+    assert
+      .dom('.hds-form-field__error')
+      .hasAttribute('id', `error-${controlId}`);
+  });
+
+  // ATTRIBUTES
```

Comment:
should there be tests for the `@layout` argument?

Reply:
@Dhaulagiri it's at line 22 of this file.